services/src/main/java/org/jboss/windup/web/services/rest/RegisteredApplicationEndpointImpl.java

```diff
@@ -20,9 +20,12 @@
     private RegisteredApplicationService registeredApplicationService;
 
     @Override
-    public Collection<RegisteredApplication> getAllApplications()
+    public Collection<RegisteredApplication> getAllApplications(Long projectId)
     {
-        return this.registeredApplicationService.getAllApplications();
+        if (projectId == null)
```

Comment:
"MigrationProject entity has applications collection" - well, it has. But jpamodelgen strips relations away.

Regarding new endpoint resource - that was my first idea too. But the way the subpaths were created, it's not possible - / is used and /{*} is used. Only after that I found I can use regex in the URL, so that could be the solution. When we get back to this task.